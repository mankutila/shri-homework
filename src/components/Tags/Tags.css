.tags {
  margin: 20px 0 5px;
  display: flex;
  flex-wrap: wrap;
}

.tag {
  cursor: pointer;
  margin: 0 15px 15px 0;
  background-color: #57906a;
  border: none;
  color: #fff;
  padding: 10px 15px;
  border-radius: 20px;
  opacity: .5;
}

.tag:focus {
  outline: none;
  box-shadow: 0 0 3px 2px rgba(0, 0, 0, 0.15)
}

.tag--current {
  opacity: 1;
}