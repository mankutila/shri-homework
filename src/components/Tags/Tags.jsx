import React, {Component} from 'react';

import {Picture} from "../Picture/Picture";

const TAGS = [
  {
    title: "Архитектура",
    slug: "architecture"
  },
  {
    title: "Океан",
    slug: "ocean"
  },
  {
    title: "Птицы",
    slug: "birds"
  }
];

export class Tags extends Component {


  render() {
    return (
      <section className="tags">
        {TAGS.map(({title, slug}, index) => (
          <button
            key={slug}
            className={`tag`}
            /*onClick={() => dispatch({
              type: 'SET_TAG',
              tag: slug
            })}*/
          >
            {title}
          </button>
        ))}
      </section>
    );
  }
}