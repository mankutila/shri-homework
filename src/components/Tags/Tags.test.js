import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import { Tags } from './Tags';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

it('renders correctly', () => {

  const testState = {
    images: {
      images: [],
      page: 1,
    },
    lightbox: {
      isOpen: false
    },
    tags: {
      current: 'architecture'
    }

  };
  const store = createMockStore(testState);

  const wrapper = shallowWithStore(<Tags />, store)

  expect(wrapper).toMatchSnapshot();
});
