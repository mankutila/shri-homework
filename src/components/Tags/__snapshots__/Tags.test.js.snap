// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TagsComponent
  current="architecture"
  dispatch={[Function]}
/>
`;
